scipy/sparse/sparsetools/bsr.h

```diff
@@ -796,13 +805,14 @@ void bsr_matvecs(const I n_brow,
     const npy_intp Y_bs = (npy_intp)n_vecs*R; //Yx blocksize
     const npy_intp X_bs = (npy_intp)C*n_vecs; //Xx blocksize
 
+    scal(n_brow * R * n_vecs, beta, Yx);
```

Comment:
Do the calls to `scal` result to loading `Y`twice?
Can the beta scaling be moved inside the loop below, for better locality --- or does this result to problems (noncanonical input)?